lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -166,7 +166,10 @@ def add_payment_method(post, payment_method, options)
         elsif payment_method.is_a?(NetworkTokenizationCreditCard)
           post[:ccnumber] = payment_method.number
           post[:ccexp] = exp_date(payment_method)
-          post[:token_cryptogram] = payment_method.payment_cryptogram
+          post[:cavv] = payment_method.payment_cryptogram
+          post[:eci] = payment_method.eci
+          post[:decrypted_applepay_data] = 1 if payment_method.source.to_s == 'apple_pay'
```

Comment:
If `payment_method.source` is a symbol you can compare it with `:apple_pay` and remove the to_s method (same with `'google_pay'`)

Reply:
thanks, updated